### ISSUE TITLE
java.lang.NoSuchFieldException: modifiers on Reflect.set() on Android

### ISSUE THREAD

**Issue by kigkrazy** [jOOQ/jOOR#48]
### Expected behavior and actual behavior:
```
Reflect.on(httpUrl).set("a", "http");
```
### Steps to reproduce the problem:
```
org.joor.ReflectException: java.lang.NoSuchFieldException: modifiers
```
### Versions:
- system : android 4.x
- jOOR: 0.9.7
- Java: 6

====

I found the problem in the code : 
```
    public Reflect set(String name, Object value) throws ReflectException {
        try {
            Field field = field0(name);
            if ((field.getModifiers() & Modifier.FINAL) == Modifier.FINAL) {
                Field modifiersField = Field.class.getDeclaredField("modifiers");//android system don't have this field.
                modifiersField.setAccessible(true);
                modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
            }
            field.set(object, unwrap(value));
            return this;
        }
        catch (Exception e) {
            throw new ReflectException(e);
        }
    }
```

**Comment by lukaseder** [comment]
Thanks for your report. How can this be done on Android? Ideally, if you could provide a working PR, that would be awesome. I cannot verify this on Android right now...

**Comment by kigkrazy** [comment]
I'm sorry to see it now.
I solve problem with that code:
```
    public Reflect set(String name, Object value) throws ReflectException {
        try {
            Field field = field0(name);
            //fix by KigKrazy
            // in anddroid, there is not "modifiers" field.
//            if ((field.getModifiers() & Modifier.FINAL) == Modifier.FINAL) {
//                Field modifiersField = Field.class.getDeclaredField("modifiers");
//                modifiersField.setAccessible(true);
//                modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
//            }
            field.set(object, unwrap(value));
            return this;
        }
        catch (Exception e) {
            throw new ReflectException(e);
        }
    }
```
my english is not good, hope U can understand.

**Comment by lukaseder** [comment]
I've committed a fix to GitHub master, due for jOOQ 0.9.8. Would you mind checking if this fixes the issue for you?
